opendatahub-io/odh-dashboard
# Remove 'opendatahub.io/link' dependency for notebook controller

Closes #844 

## Description


Currently, the notebook controller is highly depending on the `opendatahub.io/link` annotation to get the link of the Jupyter Notebook, this PR gets rid of that. Instead of using that annotation, we use `getRoute` to fetch the route of the notebook as the source of truth.

## How Has This Been Tested?



1. Search in the code base, and make sure there is no `opendatahub.io/link` anymore
2. Try to spawn a notebook using the Jupyter tile
3. Try to access the notebook
4. Try to log out of the notebook
5. Make sure everything works OK

## Test Impact


Add unit tests to test the `useNotebookRedirectLink` hook.

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
